cmd/kubeadm/app/phases/addons/addons.go

```diff
@@ -139,17 +135,28 @@ func CreateKubeDNSAddon(deploymentBytes, serviceBytes []byte, client *clientset.
 	return nil
 }
 
-// TODO: Instead of looking at the subnet given to kubeadm, it should be possible to only use /28 or larger subnets and then
-// kubeadm should look at the kubernetes service (e.g. 10.96.0.1 or 10.0.0.1) and just append a "0" at the end.
-// This way, we don't need the information about the subnet in this phase => good
-func getDNSIP(subnet string) (net.IP, error) {
-	_, n, err := net.ParseCIDR(subnet)
+// getDNSIP fetches the kubernetes service's ClusterIP and appends a "0" to it in order to get the DNS IP
+func getDNSIP(client *clientset.Clientset) (net.IP, error) {
```

Comment:
If we keep this PR open, I'm sure it will serve as a push to have someone looking into what's happening. If we have this merged, there goes the push. We should bring this up later in SIG meeting.